RUFAS/input_manager.py

```diff
@@ -2221,25 +2221,212 @@ def _validate_properties(self) -> None:
         om.add_log("Metadata properties depth", f"Max depth of metadata properties is {current_max_depth}", info_map)
         om.add_log("Metadata properties path", f"Deepest path of metadata properties is {deepest_path}", info_map)
 
-    def _metadata_number_validator(self, key_path: List[str], value: dict[str, Any]) -> None:
-        """Validator function for array type properties in metadata."""
-        pass
+    def _metadata_number_validator(self, key_path: list[str], value: dict[str, Any]) -> None:
+        """Validates number type properties in metadata."""
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._metadata_number_validator.__name__,
+        }
+        required_number_property_keys = {"type"}
+        valid_number_property_keys = {"type", "description", "minimum", "maximum", "default", "nullable"}
```

Comment:
```suggestion
        required_number_property_keys = {"type"}
        optional_number_property_keys = {"description", "minimum", "maximum", "default", "nullable"}
        valid_number_property_keys = union(required_number_property_keys, optional_number_property_keys)
```
I think this syntax would work, but I didn't check it - also there may be a cleaner way to do this. The idea though is that it would be beneficial to write out the required_number_property_keys once.

Edit: I didn't read the full code before I wrote this comment. Maybe it would be better to add the `union` step in `_validate_metadata_properties_keys` instead of writing this step into every individual validator function.

Reply:
I like adding it to the `_validate_metadata_properties_keys()` function to minimize duplicate code. And it looks like sets have their own `.union()` method for joining sets together (and automatically removing duplicates).